### PR TITLE
guest_os_booting: add new case of smbios feature

### PR DESCRIPTION
**Pull request by meinaLi**
This case is to verify the smbios element.

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.smbios_mode
JOB ID     : 0617708220a6e3e0f65168cbc50506fc86699537
JOB LOG    : /var/lib/avocado/job-results/job-2023-11-02T04.17-0617708/job.log
 (1/6) type_specific.io-github-autotest-libvirt.guest_os_booting.smbios_mode.positive_test.emulate: PASS (26.28 s)
 (2/6) type_specific.io-github-autotest-libvirt.guest_os_booting.smbios_mode.positive_test.host: PASS (23.60 s)
 (3/6) type_specific.io-github-autotest-libvirt.guest_os_booting.smbios_mode.positive_test.sysinfo: PASS (31.62 s)
 (4/6) type_specific.io-github-autotest-libvirt.guest_os_booting.smbios_mode.negative_test.invalid_mode: PASS (4.70 s)
 (5/6) type_specific.io-github-autotest-libvirt.guest_os_booting.smbios_mode.negative_test.without_sysinfo_snippet: PASS (5.17 s)
 (6/6) type_specific.io-github-autotest-libvirt.guest_os_booting.smbios_mode.negative_test.with_mismatch_uuid: PASS (4.67 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-11-02T04.17-0617708/results.html
JOB TIME   : 96.83 s
```

**Comment by meinaLi** [comment]
Depend on https://github.com/avocado-framework/avocado-vt/pull/3790.

**Comment by meinaLi** [comment]
@Yingshun @dzhengfy Could you help to review this PR? Thanks.

**Review by nanli1 — APPROVED** [review]

File: libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            system_info = get_smbios_data(entry_type, host_system)
+            expected_smbios_data = eval(params.get("expected_system_info") %
+                                        (system_info["Manufacturer"], system_info["Product Name"],
+                                         system_info["SKU Number"], system_info["Family"]))
+        return entry_type, expected_smbios_data
+
+    def compare_smbios_data(entry_type, expected_smbios_data):
+        """
+        Compare the smbios data for from guest with the guest xml
+        :params entry_type: the entry type of smbios data
```
Comment (dzhengfy) [inline libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py:86]: Please add blank line above line 86

File: libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            match = pattern.match(line)
+            if match:
+                key, value = match.groups()
+                smbios_data[key] = value
+        return smbios_data
+
+    def get_expected_smbios_data(vmxml, smbios_mode):
+        """
+        Check the hardware and firmware info in guest by dmidecode.
+        :params vmxml: the xml of guest
```
Comment (dzhengfy) [inline libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py:66]: Please add blank line above line 66

File: libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        Compare the smbios data for from guest with the guest xml
+        :params entry_type: the entry type of smbios data
+        expected_smbios_data: the expected smbios data got from guest xml
+        """
+        if entry_type == "oemStrings":
+            dmidecode_cmd = "dmidecode -t 11"
+        else:
+            dmidecode_cmd = "dmidecode -t %s" % entry_type
+        session = vm.wait_for_login()
+        #status, smbios_output = session.cmd_status_output(dmidecode_cmd)
```
Comment (dzhengfy) [inline libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py:94]: Remove this line

File: libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py:88]: Updated

File: libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py:68]: Updated

File: libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py (hunk 3 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/firmware_configuration/smbios_mode.py]: Updated

**Pull request merged**